test/spec/uri-options/proxy-options.json

```diff
@@ -0,0 +1,139 @@
+{
```

Comment:
Unfortunately our uri-options spec runner is a little under the weather at the moment (https://jira.mongodb.org/browse/NODE-3738). It might be worth taking a look at `test/integration/uri-options/uri_options.spec.test.js` to see if these tests are falling under the skip condition. I'm taking a look too, just waiting on the current run to get there.

If it is skipped we might be okay with tolerating that, but you can see in that same file I broke out tests for the recent new srv options.

Reply:
Yeah, some of the tests are being skipped here. Is it worth special-casing these tests? My understanding is that they would pass after NODE-3738.

Just to be sure – is it actually correct that the invalid connection strings have `warning: false` and `valid: false` both set? If I remove the `warning: false`, then all tests just pass here. (I based this on other spec tests which also have both set for invalid connection strings, but now I’m not sure if this was actually something I should have done or not.)

Reply:
I believe this is okay just because invalid likely takes precedence, if we're throwing then that's our warning, so we're not checking for both. Our driver started by implementing these spec tests such that connection strings that should only warn were actually throwing errors and that was considered the same. We'll try to correct this to what ever extent isn't breaking when we get to that ticket.

So good to know they pass, I think you're right we don't need to special case here then. Thanks for checking!